commands/command_status.go

```diff
@@ -224,6 +229,44 @@ func statusScanRefRange(ref *git.Ref) {
 
 }
 
+type JSONStatusEntry struct {
+	Status string `json:"status"`
+	From   string `json:"from,omitempty"`
+}
+
+type JSONStatus struct {
+	Files map[string]JSONStatusEntry `json:"files"`
+}
+
+func jsonStagedPointers(ref string) {
+	staged, unstaged, err := scanIndex(ref)
+	if err != nil {
+		ExitWithError(err)
+	}
+
+	status := JSONStatus{Files: make(map[string]JSONStatusEntry)}
+
+	for _, entry := range append(unstaged, staged...) {
+		switch entry.Status {
+		case lfs.StatusRename, lfs.StatusCopy:
+			status.Files[entry.DstName] = JSONStatusEntry{
+				Status: string(entry.Status), From: entry.SrcName,
+			}
+			break
```

Comment:
`break` is not required in Go to prevent from falling through into the `default` branch. If, however, you want to skip out of the outer `for` loop on L249, you'll need to label it and `break <label>`, i.e.,

```go
L:
        for _, entry := range append(unstaged, staged...) {
                switch entry.Status {
                case lfs.StatusRename, lfs.StatusCopy:
                        // ...
                        break L // <- will move to "HERE" below.
        }

        // HERE
```